src/eval_context.rs

```diff
@@ -1521,8 +1527,9 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
                 //let src = adt::MaybeSizedValue::sized(src);
                 //let dst = adt::MaybeSizedValue::sized(dst);
                 let src_ptr = match src {
-                    Value::ByRef(ptr) => ptr,
-                    _ => bug!("expected pointer, got {:?}", src),
+                    Value::ByRef(ptr, true) => ptr,
+                    // TODO: Is it possible for unaligned pointers to occur here?
```

Comment:
That has its own host of problems -- it creates unaligned accesses in the destructor: <https://github.com/rust-lang/rust/issues/27060#issuecomment-314964836>.